### PR TITLE
Create Footer

### LINKED ISSUE #20: Create footer
We need a footer for the website, it should follow the designs provided by the design team. Furthermore it should also be responsive.

### PR DESCRIPTION
**Pull request by pedronunes19**
Closes #20 

The footer is created as a component following the mocks.  #25 should be merged for the icons.

Screenshot from 2023-08-21 18-19-23

### CODE REVIEW

**Review by Naapperas — DISMISSED** [review]
LGTM

File: resources/js/Components/Footer.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                </a>
+                <a class="p-2.5 rounded-full flex w-fit bg-2023-red-dark" target="_blank" href="https://www.linkedin.com/company/nifeup/mycompany/">
+                    <v-icon name="io-logo-linkedin" fill="white" scale="1.3"></v-icon>
+                </a>
+                <a class="p-2.5 rounded-full flex w-fit bg-2023-teal" target="_blank" href="https://www.instagram.com/niaefeup/">
+                    <v-icon name="io-logo-instagram" fill="white" scale="1.3"></v-icon>
+                </a>
+                <a class="p-2.5 rounded-full flex w-fit bg-2023-orange" target="_blank" href="mailto:ni@aefeup.pt">
+                    <v-icon name="io-mail" fill="white" scale="1.3"></v-icon>
+                </a>
```
Comment (ttoino) [inline resources/js/Components/Footer.vue]: This repetition should be avoided with the use of components, or with a for loop

File: resources/js/Components/Footer.vue [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    <a class="p-2.5 rounded-full flex w-fit bg-2023-red-dark" target="_blank" href="https://www.linkedin.com/company/nifeup/mycompany/">
+                        <v-icon name="io-logo-linkedin" fill="white" scale="1.3"></v-icon>
+                    </a>
+                    <a class="p-2.5 rounded-full flex w-fit bg-2023-teal" target="_blank" href="https://www.instagram.com/niaefeup/">
+                        <v-icon name="io-logo-instagram" fill="white" scale="1.3"></v-icon>
+                    </a>
+                    <a class="p-2.5 rounded-full flex w-fit bg-2023-orange" target="_blank" href="mailto:ni@aefeup.pt">
+                        <v-icon name="io-mail" fill="white" scale="1.3"></v-icon>
+                    </a>    
+                </div>
```
Comment (ttoino) [inline resources/js/Components/Footer.vue]: Why are two divs necessary here?

File: resources/js/Components/Footer.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<script setup>
+</script>
+
+<template>
+    <footer class="bg-2023-bg py-4">
+        <section class="flex content-center items-center justify-center relative p-8 mx-[20%]">
+            <div class="flex gap-2 flex-row items-center w-full justify-evenly max-[900px]:hidden">
```
Comment (ttoino) [inline resources/js/Components/Footer.vue]: We should probably use tailwind's built-in breakpoints, not custom ones

**Review by ttoino — CHANGES_REQUESTED** [review]

File: resources/js/Components/Footer.vue (hunk 2 above)
Comment (pedronunes19) [reply resources/js/Components/Footer.vue]: I don't think they're strictly necessary, it was just a away of separating the styles (mostly the `grid`). Follows pretty much the same structure that was being used for the maintenance page.

Comment (ttoino) [reply resources/js/Components/Footer.vue]: While I don't really like it, it's an ok compromise as stuffing all classes in a single element would also be hard for readability

**Review by ttoino — APPROVED** [review]

File: resources/js/Components/Footer.vue [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ["io-mail", "bg-2023-orange", "mailto:ni@aefeup.pt"]
+];
+</script>
+
+<template>
+    <footer class="bg-2023-bg py-4">
+        <section class="flex content-center items-center justify-center relative p-8 mx-[25%]">
+            <div class="hidden gap-2 flex-row items-center w-full justify-evenly md:flex">
+                <template v-for="platform in contacts">
+                    <a class="p-2.5 rounded-full flex w-fit" :class="platform[1]" target="_blank" :href="platform[2]">
```
Comment (Naapperas) [inline resources/js/Components/Footer.vue:16]: The contacts could be objects, that would make the code a little cleaner. This is optional, so if you feel it does not make sense feel free to ignore this comment.

File: resources/js/Components/Footer.vue [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                    </a>
+                </template>
+            </div>
+            <div class="flex flex-col items-center gap-2 md:hidden">
+                <div class="grid grid-cols-2 gap-2 justify-center">
+                    <template v-for="platform in contacts">
+                        <a class="p-2.5 rounded-full flex w-fit" :class="platform[1]" target="_blank" :href="platform[2]">
+                            <v-icon :name="platform[0]" fill="white" scale="1.3"></v-icon>
+                        </a>
+                    </template>
```
Comment (Naapperas) [inline resources/js/Components/Footer.vue:27]: Can't we have this logic separated? That way it is not duplicated (unless I saw things wrong but I double checked and believe they are the same for the mobile and desktop styles)

**Review by Naapperas — APPROVED** [review]
LGTM! @pedronunes19 I left some suggestions that can be ignored if you believe they don't make much sense.

**Pull request merged**